addons/website/models/website_page_properties.py

```diff
@@ -0,0 +1,136 @@
+                if context_target_model_id:
+                    [target_model, target_id] = context_target_model_id.split(',')
+                    record.target_model_id = self.env[target_model].browse(int(target_id))
+                else:
+                    record.target_model_id = False
+
+    @api.depends_context('website_id')
+    def _compute_page_website_id(self):
+        for record in self:
+            if 'website_id' in record._fields:
+                record.page_website_id = record.website_id
+            if not record.page_website_id:
+                website_id = self._context.get('website_id')
+                if website_id:
+                    record.page_website_id = self.env['website'].browse(int(website_id))
+
+    @api.depends('url', 'page_website_id', 'target_model_id')
+    def _compute_menu_ids(self):
+        for record in self:
+            record.menu_ids = self.env['website.menu'].search([
+                ('website_id', '=', record.page_website_id.id),
+                ('url', '=', record.url),
+            ])
+
+    @api.depends('menu_ids')
+    def _compute_is_in_menu(self):
+        for record in self:
+            record.is_in_menu = bool(record.menu_ids)
+
+    @api.depends('menu_ids', 'url', 'page_website_id', 'target_model_id')
+    def _inverse_is_in_menu(self):
+        if self.is_in_menu:
+            if not self.menu_ids:
+                self.env['website.menu'].create({
+                    'name': self.target_model_id.name,
+                    'url': self.url,
+                    'parent_id': self.page_website_id.menu_id.id,
+                    'website_id': self.page_website_id.id,
+                })
+        elif self.menu_ids:
+            # If the page is no longer in menu, we should remove its website_menu
+            self.menu_ids.unlink()
+
+    @api.depends_context('url')
+    def _compute_url(self):
+        for record in self:
+            try:
+                record.url = self._context.get('url')
+            except RuntimeError:
+                record.url = False
+
+    @api.depends('page_website_id', 'target_model_id')
+    def _compute_is_homepage(self):
+        for record in self:
+            url = record.url
+            current_homepage_url = record.page_website_id.homepage_url or '/'
+            record.is_homepage = (
+                    url == current_homepage_url or (url == '/' and current_homepage_url != '/')
+            )
+
+    @api.depends('url', 'page_website_id', 'target_model_id')
+    def _inverse_is_homepage(self):
+        self.ensure_one()
+        url = self.url
+        if self.is_homepage:
+            if url and url != '/' and self.page_website_id.homepage_url != url:
+                self.page_website_id.homepage_url = url
+        else:
+            self.page_website_id.homepage_url = '/'
+
+    def unlink(self):
+        self.menu_ids.unlink()
+        return super().unlink()
```

Comment:
I added this to delete menus of not website.page models, which are not covered by website.menu ondelete='cascade'

It turns out that this resulted in an infinite recursive loop because of a module that I hadn't tested locally. https://runbot.odoo.com/runbot/build/67779346 : unlink website_menu > unlink website_event_menu > unlink ir_ui_view > unlink website_page_properties > unlink website_menu > ...

However, indeed `self.menu_ids.unlink()` in `website.page` is unnecessary so I'll remove it.